### PR TITLE
Support ctanglex.

### PR DESCRIPTION
**Pull request by belm0**
ctanglex can provide more readable C output.  Some minor source
changes were needed for its stricter parsing of CWEB.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,6 +3,8 @@
 CFLAGS=-fPIC -Wall -ansi -g -pedantic -O3
 SP_LDFLAGS = -lsoundpipe -lsndfile -lm
 LDFLAGS=-lsporth $(SP_LDFLAGS) -lpthread -ldl
+CTANGLE=ctangle
```
Comment (PaulBatchelor) [inline Makefile]: I would change this to the conditional assignment operator:

`CTANGLE?=ctangle`

File: Makefile [hunk 2]
```diff
@@ -3,6 +3,8 @@
...
+#CTANGLE=ctanglex +c -l
```
Comment (PaulBatchelor) [inline Makefile]: Not the biggest fan of having commented out lines with no explanation. My personal approach these days is to use variables like so:

```
ifdef USE_CTANGLEX
CTANGLE=ctanglex +c -l
endif
```

File: Makefile (hunk 1 above)
Comment (belm0) [reply Makefile]: done, thank you

File: Makefile (hunk 2 above)
Comment (belm0) [reply Makefile]: I've opted for a more descriptive comment rather than hard-coding ctanglex options and adding another variable.

**Comment by belm0** [comment]
I'm not familiar with CWEB authoring but CWEBx appears to just be some incremental feature enhancements.  Section (9) in the CWEBx manual covers the differences.

http://wwwmathlabo.univ-poitiers.fr/~maavl/CWEBx/cwebx/manual.dvi

File: Makefile (hunk 1 above)
Comment (belm0) [reply Makefile]: Actually CTANGLE is a make implicit variable.  So ?= will never get assigned :(

Implicit variables already respond to overrides, so it's sufficient to remove this line.

**Pull request merged**